### PR TITLE
Always run per-repo programs in a temporary directory.

### PR DESCRIPTION
**Pull request by ltratt**
It seems likely that nearly all per-repo programs will want to write something to disk (at the very least cloning a repository), so running them in a temporary directory (which is automatically cleaned up after the program exits) makes the programs a little less error prone and a little easier to write.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,9 @@ where:
  * `<repo-programs-dir>` is the directory where the per-repo programs are
    stored. For a repository `repo` owned by `user` the command
    `<repo-programs-dir>/<user>/<repo> <event> <path to GitHub JSON>` will be
-   run.
+   run. Note that per-repo programs are run with their current working
+   directory set to a temporary directory to which they can freely write and
+   which will be automatically removed when they have completed.
```
Comment (vext01) [inline README.md]: A nitpick. I think we can make this more easily understood with:
```
Note that per-repo programs are run in a temporary directory which will be automatically removed when they have completed.
```

I don't think it's necessary to say that the directory is writable. The user will assume this.

**Comment by vext01** [comment]
Aside from one that one small comment, LGTM.

File: README.md (hunk 1 above)
Comment (ltratt) [reply README.md]: If we don't say it's writeable, they may assume we've done some sort of `chroot` tricks and things are readable only. It certainly doesn't hurt to make it clear that users can write to this directory as they wish.

Comment (vext01) [reply README.md]: I'm not sure what chroot has to do with permissions. You can have a writable chroot.

If you want to say it's writeable, how about:
> Note that per-repo programs are run in a (writeable) temporary directory which will be automatically removed when they have completed.

Comment (ltratt) [reply README.md]: I think we might be in danger of crossing into nit-picking territory now :) I don't think there's anything inaccurate, missing, or misleading in the text.

Comment (vext01) [reply README.md]: Well, I did open with "A nitpick" :)

I think it's a bit wordy personally. I had to re-read it a couple of times to absorb it.

Compare my suggestion:
> Note that per-repo programs are run in a (writeable) temporary directory which will be automatically removed when they have completed.

With the original:
> Note that per-repo programs are run with their current working directory set to a temporary directory to which they can freely write and which will be automatically removed when they have completed.

The latter is quite jarring.

I don't feel strongly enough to argue further, mind.

Comment (ltratt) [reply README.md]: Amusingly I started with a sentence that said "run in a directory" before adding the stuff about CWD and writeable and so on. Let's keep it as it is and revise in the future.

Comment (vext01) [reply README.md]: I disagree with the decision, but so be it :P
